optimum/neuron/modeling_seq2seq.py

```diff
@@ -0,0 +1,912 @@
+            )
+
+        if task is None:
+            task = TasksManager.infer_task_from_model(cls.auto_model_class)
+
+        # Get compilation arguments
+        auto_cast_type = None if auto_cast is None else auto_cast_type
+        compiler_kwargs = {
+            "auto_cast": auto_cast,
+            "auto_cast_type": auto_cast_type,
+            "disable_fast_relayout": disable_fast_relayout,
+            "disable_fallback": disable_fallback,
+        }
+
+        save_dir = TemporaryDirectory()
+        save_dir_path = Path(save_dir.name)
+
+        main_export(
+            model_name_or_path=model_id,
+            output=save_dir_path,
+            compiler_kwargs=compiler_kwargs,
+            task=task,
+            dynamic_batch_size=dynamic_batch_size,
+            cache_dir=cache_dir,
+            trust_remote_code=trust_remote_code,
+            subfolder=subfolder,
+            revision=revision,
+            force_download=force_download,
+            local_files_only=local_files_only,
+            use_auth_token=use_auth_token,
+            do_validation=False,
+            **kwargs_shapes,
+        )
+
+        return cls._from_pretrained(
+            model_id=save_dir_path,
+            config=config,
+            model_save_dir=save_dir,
+        )
+
+    def _save_config(self, save_directory):
+        save_directory = Path(save_directory)
+        self.configs[ENCODER_NAME].save_pretrained(save_directory / ENCODER_NAME)
+        self.configs[DECODER_NAME].save_pretrained(save_directory / DECODER_NAME)
+        combined_config = self._combine_encoder_decoder_config(
+            encoder_config=self.configs[ENCODER_NAME],
+            decoder_config=self.configs[DECODER_NAME],
+        )
+        combined_config.save_pretrained(save_directory)
+
+    def _combine_encoder_decoder_config(self, encoder_config: "PretrainedConfig", decoder_config: "PretrainedConfig"):
+        encoder_neuron_config = encoder_config.neuron
+        decoder_neuron_config = decoder_config.neuron
+        combined_config = copy.deepcopy(encoder_config)
+
+        encoder_neuron_config["encoder_input_names"] = encoder_neuron_config.pop("input_names")
+        encoder_neuron_config["encoder_output_names"] = encoder_neuron_config.pop("output_names")
+        decoder_neuron_config["decoder_input_names"] = decoder_neuron_config.pop("input_names")
+        decoder_neuron_config["decoder_output_names"] = decoder_neuron_config.pop("output_names")
+
+        encoder_neuron_config.update(decoder_neuron_config)
+        encoder_neuron_config.pop("model_type")
+        combined_config.__setattr__("neuron", encoder_neuron_config)
+
+        return combined_config
+
+    def can_generate(self):
+        logger.warning(
+            "NeuronModelForConditionalGeneration is an abstract class and is not meant to be used for generation. Please use NeuronModelForSeq2SeqLM instead."
+        )
```

Comment:
Why warn when the method is used to check if it is possible?
It should throw a warning if the users tries to call generate instead no?

Reply:
I will remove this method and turn this class to an abstract class instead.